sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -340,7 +340,8 @@ class Dataset[T] private[sql](
    */
   // This is declared with parentheses to prevent the Scala compiler from treating
   // `ds.toDF("1")` as invoking this toDF and then apply on the returned DataFrame.
-  def toDF(): DataFrame = new Dataset[Row](sparkSession, queryExecution, RowEncoder(schema))
+  def toDF(): DataFrame =
+    new Dataset[Row](sparkSession, queryExecution, RowEncoder(logicalPlan.schema))
```

Comment:
Actually I think we should do this. Otherwise `ds.toDF()` may have different column order from `ds`, and this breaks the commonly used `checkDataset` testing method, which always converts a Dataset to a DataFrame first and then checks output rows of the DataFrame.